cmd/install.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/k0sproject/k0s/pkg/apis/v1beta1"
+	"github.com/k0sproject/k0s/pkg/install"
+)
+
+func init() {
+	installCmd.Flags().StringVar(&role, "role", "server", "node role (possible values: server or worker. In a single-node setup, a worker role should be used)")
+}
+
+var (
+	role string
+
+	installCmd = &cobra.Command{
+		Use:   "install",
+		Short: "Helper command for setting up k0s on a brand-new system. Must be run as root (or with sudo)",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			switch role {
+			case "server", "worker":
+				return setup()
+			default:
+				logrus.Errorf("invalid value %s for install role", role)
+				return cmd.Help()
+			}
+		},
+	}
+)
+
+// the setup functions:
+// * Ensures that the proper users are created
+// * sets up startup and logging for k0s
+func setup() error {
+	if os.Geteuid() != 0 {
+		logrus.Fatal("this command must be run as root!")
+	}
+
+	if role == "server" {
+		if err := createServerUsers(); err != nil {
+			logrus.Errorf("failed to create server users: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func createServerUsers() error {
+	clusterConfig, err := ConfigFromYaml(cfgFile)
+	if err != nil {
+		return err
+	}
+	users := getUserList(*clusterConfig.Install.SystemUsers)
```

Comment:
If the config yaml file doesn't contain `installConfig:` the command will fail with:

```
# sudo ./k0s install --config cfg.yaml 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1aef50d]

goroutine 1 [running]:
github.com/k0sproject/k0s/cmd.createServerUsers(0x6b, 0x0)
        /root/src/github.com/k0sproject/k0s/cmd/install.go:60 +0x6d
...
```
Maybe we should check if value of system users are present

Reply:
Resolved  with https://github.com/k0sproject/k0s/blob/a93e731e3652f53123cd224c01064dd78b7f811c/cmd/helpers.go#L47-L49

Reply:
yes it is resolved